src/app.jsx

```diff
@@ -89,98 +89,39 @@ export const Application = () => {
-                            setFiles(_files);
-                            setLoadingFiles(false);
-                        });
+    useEffect(
+        () => {
+            if (sel === undefined) {
+                return;
             }
-        });
-    }, [currentDir]);
-
-    const watchFiles = useCallback(() => {
-        if (channel.current !== null)
-            channel.current.close();
-
-        channel.current = cockpit.channel({
-            payload: "fswatch1",
-            path: `/${currentDir}`,
-            superuser: "try",
-        });
-
-        channel.current.addEventListener("message", (ev, data) => {
-            const item = JSON.parse(data);
 
-            item.name = item.path.slice(item.path.lastIndexOf("/") + 1);
-            item.isHidden = item.name.startsWith(".");
-
-            // When files are created with some file editor we get also 'attribute-changed' and
-            // 'done-hint' events which are handled below. We should not add the same file twice.
-            if (item.event === "created" && item.type === "directory") {
-                updateFile(item, currentDir).then(file => {
-                    setFiles(_f => [..._f, file]);
-                });
-            } else {
-                if (item.event === "deleted") {
-                    setFiles(f => f.filter(res => res.name !== item.name));
-                } else {
-                    // For events other than 'present' we don't receive file stat information
-                    // so we rerun the fslist command to get the updated information
-                    // https://github.com/allisonkarlitskaya/systemd_ctypes/issues/56
-                    if (item.name[0] !== ".") {
-                        getFsList();
-                    }
-                }
-            }
-        });
-    }, [currentDir, getFsList]);
-
-    useEffect(() => {
-        // Wait for the path initial value to be set before fetching the files
-        if (sel === undefined) {
-            return;
-        }
-
-        setSelected([]);
-        setFiles([]);
-        setLoading(true);
-
-        watchFiles();
-        getFsList();
-    }, [sel, getFsList, watchFiles]);
+            console.log('create client for', currentDir);
+            const attrs = ["type", "mode", "mtime", "group", "user", "size", "target"];
+            const info = fsinfo(`/${currentDir}/`, attrs, { fnmatch: '*', targets: 'stat' });
+            return info.effect(state => {
+                console.log('state update start');
+                setLoading(false);
+                setLoadingFiles(!(state.info || state.error));
+                setRootInfo(state.info);
+                setErrorMessage(state.error?.message ?? "");
+                const entries = Object.entries(state?.info?.entries || {});
+                const files = entries.map(([name, attrs]) => ({
+                    ...attrs,
+                    name,
+                    to: info.target(name)?.type ?? null
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-210-20240129-115925-92bcde33-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)